tests/test_constraints.py

```diff
@@ -933,6 +933,61 @@ async def test_constraints_ddl_06(self):
                 };
             """)
 
+    async def test_constraints_ddl_07(self):
+        await self.con.execute("""
+            SET MODULE test;
+            CREATE TYPE Test {
+                CREATE PROPERTY first_name -> str;
+                CREATE PROPERTY last_name -> str;
+                CREATE CONSTRAINT exclusive on (__subject__.first_name);
```

Comment:
Make sure that only `SINGLE` pointers are allowed, and test constraints on links too.

Reply:
Per chat, we'll allow `MULTI` if it's only one ref in expression, so that it's completely equivalent to a constraint defined directly on a pointer.